SameDayReid/gallery_creation.py

```diff
@@ -70,46 +85,121 @@ def add_video_to_gallery(self, video_path:str, face_clf, skip_every=500):
             confs = result['track_results'][0][:, -1]
             crops_bboxes = result['track_results'][0][:, 1:-1]
             crops_imgs = mmcv.image.imcrop(img, crops_bboxes, scale=1.0, pad_fill=None)
-            crop_candidates_inds = []
-            crop_candidates_faces = []
-            for i, (id, conf, crop_im) in enumerate(zip(ids, confs, crops_imgs)):
-                try:
-                    ## first detection run --- detect all faces in image
-                    face_imgs, face_bboxes, face_probs = self.arc.detect_face_from_img(crop_img=crop_im)
-                    # if only a single face exists, apply a high threshold to get a high resolution face image and body
-                    if face_imgs is not None and len(face_imgs) == 1:
-                        face_img, face_prob = self.pose_estimator.find_matching_face(crop_im, face_bboxes, face_probs,
-                                                                                    face_imgs)
-                        if is_img(face_img) and face_prob >= 0.8: # hard coded min threshold for face rec
-                            crop_candidates_faces.append(face_img)
-                            crop_candidates_inds.append(i)
-                        # else: TODO write when pose estimation discards image
-
-                except Exception as e:
-                    # print(e)
-                    # print('err')
-                    continue
-            crop_cands = [crop_im for i, crop_im in enumerate(crops_imgs) if i in crop_candidates_inds]
-            if len(crop_candidates_faces) > 0: # some faces where detected
-                for i, (face,crop_im) in enumerate(zip(crop_candidates_faces, crop_cands)):
-                    # numpy_img = face.permute(1, 2, 0).int().numpy().astype(np.uint8)
-                    face = face[:, :, ::-1]
+            video_crops.extend(self.detect_faces_and_create_db_crops(ids=ids,
+                                                        confs=confs,
+                                                        crops_imgs=crops_imgs,
+                                                        crops_bboxes=crops_bboxes,
+                                                        vid_name=vid_name,
+                                                        part=part,
+                                                        frame_num=image_index
+                                                        ))
+
+        add_entries(crops=video_crops, db_location=SAME_DAY_DB_LOCATION) # Note this points to the SAME_DB_DB_LOCATION !
```

Comment:
If you think we will use this function somewhere else with a different db_location, then set the db_location as a parameter that defaults to SAME_DAY_SB_LOCATION. If not, that just leave it for now as is.

Reply:
Not sure I follow --- we do use it in a different location (when we build the actual labelled gallery for val-test videos), here we just call it with a different db_location as this is the SAME_DAY_DB. 
In this case, I even suggest removing the default db_location of add-entires (in dal.py) and setting it as a required param. WDYT?

Reply:
Sorry, my comment was unclear. I mean the function that you are currently in ( add_video_to_db()). If we will use this function with db_locations, then let's set it as a parameter to the function. Otherwise, we will need to change this in the code every time

Reply:
Ah I see what you mean. No, currently we will not call it on any other DB other then this one, unless we change dataset, I will add it as a param in case we will